source/js/donate-modal/donate-modal.jsx

```diff
@@ -2,58 +2,47 @@ import React from "react";
 import ReactDOM from "react-dom";
 import ReactGA from "../react-ga-proxy.js";
 
-const KEY_STATE = `donate modal state`;
-const KEY_TIMER = `donate modal timer`;
-const DELAY = 10000; // in ms
-const TIMER_INCREMENT = 1000; // in ms
+const DISMISSED_CHECK_KEY = "fundraising-banner";
+const MILLISECONDS = 1000;
+const DAY = 24 * 3600 * MILLISECONDS;
+const localStorage = typeof window === "undefined" ? {} : window.localStorage;
 
 class DonateModal extends React.Component {
   constructor(props) {
     super(props);
 
-    let dismissed = !!sessionStorage.getItem(KEY_STATE) || false;
-    let prevTimer = this.getPrevTimer();
+    const INITIAL_DELAY = parseInt(this.props.delay) || 10 * MILLISECONDS; // 10 seconds, in ms
+    const DISMISSAL_DELAY = parseInt(this.props.hideFor) || 7; // one week, in days
+
+    // Don't treat the modal as dismissed if:
+    // 1. the user never dismissed it before, or
+    // 2. it's been longer than a week since they dismissed
+    let dismissed = false;
+    let lastLoad = localStorage[DISMISSED_CHECK_KEY];
+    if (lastLoad) {
+      var diff = (Date.now() - parseInt(lastLoad, 10)) / DAY;
+      if (diff < DISMISSAL_DELAY) {
+        dismissed = true;
+      }
+    }
 
     this.state = {
-      delay: DELAY - prevTimer,
+      delay: INITIAL_DELAY,
```

Comment:
if value of `delay` won't be changed, should we turn it into a class property instead a state?

Reply:
it can be changed through a prop, though.